### PR TITLE
add usecontext and useReducers to TODO

### PR DESCRIPTION
**Pull request by AhmadMaree**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -15,12 +15,13 @@
...
         },
-        "ecmaVersion": 12,
+        "ecmaVersion": 2020,
         "sourceType": "module"
     },
     "plugins": [
         "react"
     ],
     "rules": {
+            "react/prop-types":"off"
```
Comment (HinnHabib) [inline .eslintrc.json:25]: We need to enable this rule

File: src/container/ChartTodo/ChartTodo.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                       item[value.Date] =  item[value.Date]|| {Date:value.Date,count: 0};
+                       item[value.Date].count++;
+                       return item;
+                   },[])
+                   setTodoList(todoData)
+                   setLoading(false)
+           }).catch(err=>{
+                  setError(err)
+                  setLoading(false)   
+           })
```
Comment (HinnHabib) [inline src/container/ChartTodo/ChartTodo.js:29]: No need to repeat setLoading to false. As we can use the finally keyward

File: src/container/ChartTodo/ChartTodo.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                  setError(err)
+                  setLoading(false)   
+           })  
+     },[])
+
+
+        let chart = <CircularProgress className={classes.Loading} />
+        if(!loading) {
+            if(error) {
+            chart = <p style={{textAlign : 'center',fontSize :30}}>{error.message}</p>
```
Comment (HinnHabib) [inline src/container/ChartTodo/ChartTodo.js:36]: Please used the typography component from the material 
https://material-ui.com/components/typography/#typography

Comment (HinnHabib) [reply src/container/ChartTodo/ChartTodo.js:36]: and I don't recommend using inline-style

File: src/context/Provider.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React , {useReducer,createContext} from 'react';
+import initaltodoState from './initalStates/Todo';
+import todoReducer from './reducers/Todo';
+
+
+export const GlobalContext = createContext();
+
+export const GlobalContextProvider = props => {
```
Comment (HinnHabib) [inline src/context/Provider.js:8]: Not sure global context is a good name I perfer to use a more specified name

**Review by HinnHabib — CHANGES_REQUESTED** [review]

File: src/container/ChartTodo/ChartTodo.js (hunk 2 above)
Comment (AhmadMaree) [reply src/container/ChartTodo/ChartTodo.js:29]: Ya true

**Pull request closed without merging**